xml_converter/src/packaging_protobin.cpp

```diff
@@ -235,6 +234,8 @@ void write_protobuf_file(
         }
     }
 
+    StringHierarchy category_filter;
+    category_filter.add_path({}, true);
```

Comment:
Why are you hardcoding the global wildcard here and preventing it from being passed in to the function?

Reply:
I believe that category_filter is not global in the same sense as the other two large inputs. It is created just before this function is called, and always for a single path. In this sense, it is more similar in scope to the `ProtoWriterState`, which is also created in this function.